services/streammanager/kafka/kafkamanager.go

```diff
@@ -450,14 +451,21 @@ func serializeAvroMessage(value []byte, codec goavro.Codec) ([]byte, error) {
 	return binary, nil
 }
 
-func prepareBatchOfMessages(topic string, batch []map[string]interface{}, timestamp time.Time, p producerManager) (
+func prepareBatchOfMessages(batch []map[string]interface{}, timestamp time.Time, p producerManager) (
 	[]client.Message, error,
 ) {
 	start := now()
 	defer func() { kafkaStats.prepareBatchTime.SendTiming(since(start)) }()
-
 	var messages []client.Message
+
 	for i, data := range batch {
+		topic := data["topic"].(string)
```

Comment:
Let's not be this relaxed and try to avoid panics:

```go
topic, ok := data["topic"].(string)
if !ok {
    // handle error
}
```

Reply:
This is addressed